ra2ce/analysis/adaptation/adaptation.py

```diff
@@ -99,18 +104,25 @@ def run_benefit(self) -> GeoDataFrame:
 
         return self.adaptation_collection.calculation_options_benefit(_benefit_gdf)
 
-    def calculate_bc_ratio(self) -> GeoDataFrame:
+    def calculate_bc_ratio(
+        self, benefit_gdf: GeoDataFrame, cost_gdf: GeoDataFrame
+    ) -> GeoDataFrame:
         """
         Calculate the benefit-cost ratio for all adaptation options.
 
+        Args:
+            benefit_gdf (GeoDataFrame): Gdf containing the benefit of the adaptation options.
+            cost_gdf (GeoDataFrame): Gdf containing the cost of the adaptation options.
+
         Returns:
-            GeoDataFrame: The result of the benefit-cost ratio calculation.
+            GeoDataFrame: Gdf containing the benefit-cost ratio of the adaptation options.
         """
-        _cost_gdf = self.run_cost()
-        _benefit_gdf = self.run_benefit()
-
-        # TODO: apply economic discounting
-        # TODO: calculate B/C ratio
-        # TODO: apply overlay
+        for _option in self.adaptation_collection.adaptation_options:
+            benefit_gdf[f"{_option.id}_cost"] = cost_gdf[f"{_option.id}_cost"]
+            benefit_gdf[f"{_option.id}_bc_ratio"] = benefit_gdf[
+                f"{_option.id}_benefit"
+            ].replace(float("nan"), 0) / benefit_gdf[f"{_option.id}_cost"].replace(
```

Comment:
it should be cost / benefit.
if a benefit is 0 for a link (because Impact_0 = Impact_ID) then assign CB ratio ro Nan